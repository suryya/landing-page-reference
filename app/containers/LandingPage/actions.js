import { REHYDRATE } from 'redux-persist/lib/constants';

export const FETCH_EMPLOYEES = '@@employees/FETCH_EMPLOYEES';
export const FETCH_EMPLOYEES_SUCCESS = '@@employees/FETCH_EMPLOYEES_SUCCESS';
export const FETCH_EMPLOYEES_ERROR = '@@employees/FETCH_EMPLOYEES_ERROR';
export const FETCH_SINGLE_EMPLOYEE = '@@employees/FETCH_SINGLE_EMPLOYEE';
export const FETCH_SINGLE_EMPLOYEE_SUCCESS = '@@employees/FETCH_SINGLE_EMPLOYEE_SUCCESS';
export const FETCH_SINGLE_EMPLOYEE_ERROR = '@@employees/FETCH_SINGLE_EMPLOYEE_ERROR';

export const employeeListReset = () => ({
  type: REHYDRATE,
  payload: {},
});

export const fetchSingleEmployee = (payload) => ({
  type: FETCH_SINGLE_EMPLOYEE,
  payload,
});

export const fetchSingleEmployeeSuccess = (payload) => ({
  type: FETCH_SINGLE_EMPLOYEE_SUCCESS,
  payload,
});

export const fetchSingleEmployeeError = (payload) => ({
  type: FETCH_SINGLE_EMPLOYEE_ERROR,
  payload,
});

export const fetchEmployees = (payload) => ({
  type: FETCH_EMPLOYEES,
  payload,
});

export const fetchEmployeesSuccess = (payload) => ({
  type: FETCH_EMPLOYEES_SUCCESS,
  payload,
});

export const fetchEmployeesError = (payload) => ({
  type: FETCH_EMPLOYEES_ERROR,
  payload,
});
