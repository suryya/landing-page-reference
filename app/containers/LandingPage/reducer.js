/* eslint-disable */
//import { handleActions } from 'redux-actions';
import immutable from 'immutability-helper';
import { REHYDRATE } from 'redux-persist/lib/constants';
import isEmpty from 'lodash/isEmpty';

/// LOGIN_USER_DATA, LOGIN_USER_DATA_SUCCESS, LOGIN_USER_DATA_ERROR , 
// Action types
import {
  FETCH_EMPLOYEES, FETCH_EMPLOYEES_SUCCESS, FETCH_EMPLOYEES_ERROR,
  FETCH_SINGLE_EMPLOYEE, FETCH_SINGLE_EMPLOYEE_ERROR, FETCH_SINGLE_EMPLOYEE_SUCCESS
} from './actions';

export const employeeState = {
  success: false,
  loading: false,
  error: null,
  loaded: false,
  list:[],
  single:null
};


export default function employeeReducer(stateVal = employeeState, action) {

  const stateAactions = {
    [REHYDRATE]: (state) => immutable(state, {
      success: { $set: false },
      loading: { $set: false },
      error: { $set: null },
      loaded: { $set: false },
      list:  { $set: [] },
      single:{}
    }),
    [FETCH_EMPLOYEES]: (state) => immutable(state, {
      loading: { $set: true },
      success: { $set: false },
      loaded: { $set: false }
    }),
    [FETCH_EMPLOYEES_SUCCESS]: (state, { payload }) => {
      return immutable(state, {
        list: { $set: (payload) || [] },
        loading: { $set: false },
        success: { $set: true },
        error: { $set: null },
        loaded: { $set: true }
      });
    },
    [FETCH_EMPLOYEES_ERROR]: (state, { payload }) => {
      let err = (payload && !isEmpty(payload.error) && payload.error) || '';
      err = !isEmpty(err) ? err : (payload && payload.msg) || '';
      err = !err || 'Error'
      // const emailErr = /email/g.test(err) ? err : '';
      // const passwordErr = /password/g.test(err) ? err : '';

      return immutable(state, {
        loading: { $set: false },
        error: { $set: { message: err} },
        success: { $set: false },
        loaded: { $set: true }
      });
    },
    [FETCH_SINGLE_EMPLOYEE]: (state) => immutable(state, {
      loading: { $set: true },
      success: { $set: false },
      loaded: { $set: false }
    }),
    [FETCH_SINGLE_EMPLOYEE_SUCCESS]: (state, { payload }) => {
      return immutable(state, {
        single: { $set: (payload) || [] },
        loading: { $set: false },
        success: { $set: true },
        error: { $set: null },
        loaded: { $set: true }
      });
    },
    [FETCH_SINGLE_EMPLOYEE_ERROR]: (state, { payload }) => {
      let err = (payload && !isEmpty(payload.error) && payload.error) || '';
      err = !isEmpty(err) ? err : (payload && payload.msg) || '';
      err = !err || 'Error'
      // const emailErr = /email/g.test(err) ? err : '';
      // const passwordErr = /password/g.test(err) ? err : '';

      return immutable(state, {
        loading: { $set: false },
        error: { $set: { message: err} },
        success: { $set: false },
        loaded: { $set: true }
      });
    },

  }
  return stateAactions[action.type] ? stateAactions[action.type](stateVal, action) : stateVal;
}