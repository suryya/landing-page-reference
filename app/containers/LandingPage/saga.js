/* eslint-disable */
//import { takeLatest, put, call, all } 
import * as SAGA from 'redux-saga/effects';
import {
  FETCH_EMPLOYEES,
  FETCH_SINGLE_EMPLOYEE,
  fetchEmployeesSuccess,
  fetchEmployeesError,
  fetchSingleEmployeeSuccess,
  fetchSingleEmployeeError,
} from './actions';

import * as Rest from '../../remote/restMethods';


export function* singleeEployeeDataFetch(action, fetchSingleEmployeesDetails = Rest.fetchSingleEmployeesDetails ) {
  try {
    const res = yield SAGA.call(fetchSingleEmployeesDetails, action.payload);
    
    yield SAGA.put(fetchSingleEmployeeSuccess(res.data));
  } catch (err) {
    yield SAGA.put(fetchSingleEmployeeError('Something went wrong!',err));
  }
}

export function* employeeDataFetch(action, fetchEmployeesBasic = Rest.fetchEmployeesBasic ) {
  try {
    const res = yield SAGA.call(fetchEmployeesBasic, {});
    
    yield SAGA.put(fetchEmployeesSuccess(res.data));
  } catch (err) {
    yield SAGA.put(fetchEmployeesError('Something went wrong!'));
  }
}



// Watcher Saga
export function* watchFetchEmployeeData() {
  yield SAGA.takeLatest(FETCH_EMPLOYEES, employeeDataFetch);
  yield SAGA.takeLatest(FETCH_SINGLE_EMPLOYEE, singleeEployeeDataFetch);
}

export default function* employeeSaga() {
  yield SAGA.all([
    watchFetchEmployeeData()
  ]);
}
