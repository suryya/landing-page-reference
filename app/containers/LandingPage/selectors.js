/* eslint-disable */
import { createSelector } from 'reselect';


const selectEmployees = (state) => {
    
  return ( state && state.employees && state.employees.list) ? state.employees.list : []
}

const selectSingleEmployee = (state) => {
    
  return ( state && state.employees && state.employees.single) ? state.employees.single : null
}

const makeSelectEmployeesStatus = () => createSelector(
  selectEmployees,
  (employees) => {
    return employees || null
    }
);

const makeSelectSingleEmployeeStatus = () => createSelector(
  selectSingleEmployee,
  (single) => {
    return single || null
    }
);

export {
  selectEmployees,
  selectSingleEmployee,
  makeSelectEmployeesStatus,
  makeSelectSingleEmployeeStatus
};
