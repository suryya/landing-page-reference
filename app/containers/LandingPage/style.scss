.not-found-page {
  h1 {
    font-size: 2em;
    margin-bottom: 0.25em;
  }
}

.centered {
  width: 100%;
  height: 100%;
}