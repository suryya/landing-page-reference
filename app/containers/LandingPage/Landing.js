/* eslint-disable */
/**
 * LandingPage
 *
 * This is the page we show when the user visits a url that doesn't have a route
 */


import React , { useEffect, useState } from 'react';

import {Client as Styletron} from 'styletron-engine-atomic';
import {Provider as StyletronProvider} from 'styletron-react';
import {LightTheme, BaseProvider, styled, useStyletron, DarkTheme, lightThemePrimitives, createTheme} from 'baseui';


import {Block} from 'baseui/block';

import './style.scss';

import ChevronRight from 'baseui/icon/chevron-right';
import {StatefulMenu} from 'baseui/menu';
import {Heading, HeadingLevel} from 'baseui/heading';
import LoadingIndicator from 'components/LoadingIndicator';

import {
  ListItem,
  ListItemLabel,
  MenuAdapter,
  ARTWORK_SIZES,
} from 'baseui/list';


const engine = new Styletron();
const Centered = styled('div',  ({$theme}) =>  ({
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  height: '60px',
  width: '100%',
  position: 'fixed',
  zIndex: 10,
  background: $theme.colors.accent100,
}));

const Caption = styled('div',  ({$theme}) =>  ({
  display: 'block',
  justifyContent: 'center',
  alignItems: 'center',
  height: '30px',
  width: '100%',
  background: $theme.colors.primary50,
}));

const PageContainer = styled('div',  ({$theme}) =>  ({
  display: 'flex', 
  height:'100%',
  width:'100%',
  position:'absolute'
}));

const ContentContainer = styled('div',  ({$theme}) =>  ({
    display: 'flex', width: '100%', width: '100%' , marginTop:'62px'}));

const FullWidthBlockDiv = styled('div',  () =>  ({
  display: 'block',
  width: '100%'
}));

const ContentColumnOne = styled('div',  ({$theme}) =>  ({
   display: 'flex', flex: '1 1 0%'}));

const ContentColumnTwo = styled('div',  ({$theme}) =>  ({
    margin: '0px 8px', display: 'flex',  flex: '1 1 0%'}));
 

const itemProps = {
  height: 'scale1000',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
};

const wideItemProps = {
  ...itemProps,
  overrides: {
    Block: {
      style: ({$theme}) => ({
        width: `calc((200% - ${$theme.sizing.scale800}) / 3)`,
      }),
    },
  },
};

const Spaned = styled('span', ({$theme}) => ({
  display: 'inline-flex',
  fontSize: $theme.sizing.scale500,
}));

const SpanedLargefonts = styled('span', ({$theme}) => ({
  display: 'inline-flex',
  fontSize: $theme.sizing.scale600,
  fontWeight: 600
}));

const SpanedBlock = styled('span', ({$theme}) => ({
  display: 'block',
}));

const MenuContainerDiv = styled('div', ({$theme}) => ({
  marginTop: '9px'
}));

export const EmployeeList = (props) => {

  const { employees , fetchSingleEmployee} = props;

  return (
    <StatefulMenu
    items={employees}
    onItemSelect={(data) => fetchSingleEmployee(data.item.id)}
    overrides={{
      List: {
        style: {
          width: '100%',
        },
      },
      Option: {
        props: {
          overrides: {
          ListItem: {
            component: React.forwardRef((props, ref) => {


              return (<MenuAdapter
                      {...props}
                      ref={ref}
                      artwork={props.item.icon}
                      artworkSize={ARTWORK_SIZES.LARGE}
                      endEnhancer={() => <ChevronRight />}
                      > 
                  
                      <MenuContainerDiv data-testid="employee-record">
                        <SpanedLargefonts>
                        {props.item.name}
                        </SpanedLargefonts>
                        <SpanedBlock> 
                        {`Age: ${props.item.age} years`}
                        {`, `}
                        {`Employee Id: ${props.item.employeeId}`}
                        </SpanedBlock>
                      </MenuContainerDiv>
                      </MenuAdapter>)
            }),
          },
        },
        },
      },
    }}
  />
  );
}

export const Addresses = (props) => {
  const [css] = useStyletron();

  const { employee , loading} = props;
  const { addresses , name , employeeId } = employee
  return (
    <>
      {(addresses && addresses.length == 0) ? 
         (<FullWidthBlockDiv data-testid="employee-addresses">
            <FullWidthBlockDiv>
                <Caption>
                  <Block><b>{`${name}'s`}</b> doesn't have any address</Block>
                </Caption>
            </FullWidthBlockDiv>
         </FullWidthBlockDiv>) : null}
     
      {(addresses && addresses.length > 0) ? 
         (<FullWidthBlockDiv data-testid="employee-addresses">
          <FullWidthBlockDiv>
                
              <Caption>
                <Block><b>{`${name}'s`}</b> Addresses are following:</Block>
              </Caption>
            
              <ul
              className={css({
                width: '100%',
                height: 'auto',
                display: 'block',
                paddingLeft: 0,
                paddingRight: 0,
                marginTop:0
              })}
              >      
                {(addresses.map((addr) => {
                  return (
                          <React.Fragment key={addr.id} >
                          <ListItem >
                            <ListItemLabel>
                            <span data-testid="employee-address-location">{addr.location}</span>
                            </ListItemLabel>
                          </ListItem>
                          </React.Fragment>
                          )
                }))}
              </ul>  
          </FullWidthBlockDiv> 
        </FullWidthBlockDiv>) : null
      }
    </>
    
  )
};

export default function LandingPage (props) {

  const { fetchEmployees, fetchSingleEmployee , 
          employees , singleSelectedEmployee,
          loading } = props;

  useEffect(
    () => {
      fetchEmployees()
    },
    []
  )
  
  return (
    <section className="centered" >
      
      <StyletronProvider value={engine}>
        <BaseProvider theme={LightTheme}>
        <> 
          <HeadingLevel>
            <Centered>
              <Heading data-testid="list-caption"  styleLevel={4}>Customer List</Heading>
            </Centered>
          </HeadingLevel>

          <PageContainer>
            
            <ContentContainer>
            
              <ContentColumnOne data-testid="employee-list" >           
                  { (employees && employees.length <= 0) ? <LoadingIndicator/> : null}
                  { (employees && employees.length > 0) ? 
                    <EmployeeList employees={employees}  fetchSingleEmployee={fetchSingleEmployee} /> : null
                  }
              </ContentColumnOne>

              <ContentColumnTwo>
                  {( (!singleSelectedEmployee) && !loading) ?
                    (<Caption>{`Select An employee in the left column to see the employee\'s address`}</Caption>) : null
                  }

                  {( !!loading) ? (<LoadingIndicator/>) : null}


                  {singleSelectedEmployee ? <Addresses employee={singleSelectedEmployee}/> : null}

              </ContentColumnTwo>

            </ContentContainer>
          </PageContainer>
        </>
        </BaseProvider>
      </StyletronProvider>
    </section>
  );
}