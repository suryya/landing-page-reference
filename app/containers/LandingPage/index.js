import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';
import injectReducer from 'utils/injectReducer';
import injectSaga from 'utils/injectSaga';
import {
  makeSelectLoading,
  makeSelectError,
} from 'containers/App/selectors';

import {
  makeSelectEmployeesStatus,
  makeSelectSingleEmployeeStatus,
} from './selectors';
import reducer from './reducer';
import saga from './saga';
import Landing from './Landing';
import { fetchEmployees, fetchSingleEmployee } from './actions';

const mapDispatchToProps = (dispatch) => ({
  fetchSingleEmployee: (id) => {
    dispatch(fetchSingleEmployee(id));
  },
  fetchEmployees: () => {
    dispatch(fetchEmployees());
  }
});

const mapStateToProps = createStructuredSelector({
  loading: makeSelectLoading(),
  error: makeSelectError(),
  employees: makeSelectEmployeesStatus(),
  singleSelectedEmployee: makeSelectSingleEmployeeStatus(),
});

const withConnect = connect(mapStateToProps, mapDispatchToProps);

const withReducer = injectReducer({ key: 'employees', reducer });
const withSaga = injectSaga({ key: 'employees', saga });

export default compose(withReducer, withSaga, withConnect)(Landing);
export { mapDispatchToProps };
