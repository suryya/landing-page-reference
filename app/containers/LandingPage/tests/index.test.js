/**
 * Testing the LandinPage
 */
/* eslint-disable */
import React from 'react';
import { shallow, mount } from 'enzyme';
// import { Block } from 'baseui/block';
// import { Provider as StyletronProvider } from 'styletron-react';
import { BaseProvider } from 'baseui';

// import LoginDialog from '../index';

import LandingPage from '../Landing';
import history from '../../../utils/history';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { createStore } from 'redux';
import { render, fireEvent, cleanup, waitForElement } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

const employeeSuccessResp = [{
  id: 1,
  employeeId: 'Emp123',
  name: 'john dow',
  age: 18,
  addresses: [
    {
      id: 1,
      location: 'Kolkata'
    },
    {
      id: 2,
      location: 'Delhi'
    },
    {
      id: 3,
      location: 'Bombay'
    },
    {
      id: 4,
      location: 'Hydrabad'
    }
  ]
}];


let functions = {
  fetchEmployees: jest.fn().mockImplementation(() => null),
  fetchSingleEmployee: jest.fn().mockImplementation(() => null)
};

let data = {
  employees: employeeSuccessResp,
  singleSelectedEmployee: employeeSuccessResp[0]
};

afterEach(cleanup);

describe('<LandingPage/> loaded', () => {
  
  it('Renders without crashing', () => {
    const { getByTestId, getByText, queryAllByTestId } = render(
      <LandingPage {...functions} {...data}/>,
    );
    //console.log(queryAllByTestId("filter-category").length)
    expect(getByTestId('list-caption')).toHaveTextContent('Customer List');
    expect(queryAllByTestId('employee-record').length).toBe(1);
    expect(queryAllByTestId('employee-list').length).toBe(1); //.toBeGreaterThan(1)
    expect(functions.fetchEmployees).toHaveBeenCalled();
  });


  it('Clicking record shows employee addresses', () => {
    const { getByTestId, getByText, queryAllByTestId } = render(
      <LandingPage {...functions} {...data}/>,
    );

    let employeeRecords = queryAllByTestId('employee-record');
    fireEvent.click(employeeRecords[0]);
    expect(queryAllByTestId('employee-addresses').length).toBe(1);    
    expect(queryAllByTestId('employee-address-location').length).toBe(4);    

    
  });  

});


describe('<LandingPage /> snapshot', () => {
  let store;
  let component;

  beforeEach(() => {
    
    component = renderer.create(
          <LandingPage {...functions}  {...data}/>
    );
  });

  it('should render the component and compares the snapshot with old snapshot', () => {
    expect(component).toMatchSnapshot();
  });

});
