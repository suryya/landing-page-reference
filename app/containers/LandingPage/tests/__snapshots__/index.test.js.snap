// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> snapshot should render the component and compares the snapshot with old snapshot 1`] = `
<section
  className="centered"
>
  <div
    className=""
  >
    <div
      className="ae af ag ah ai aj ak al"
    >
      <h1
        className="am an ao ap aq"
        data-baseweb="heading"
        data-testid="list-caption"
      >
        Customer List
      </h1>
    </div>
    <div
      className="ae ar ai as"
    >
      <div
        className="ae ai at"
      >
        <div
          className="ae au"
          data-testid="employee-list"
        >
          <ul
            aria-activedescendant={null}
            className="av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8 b9 ba bb ai"
            data-baseweb="menu"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            role="listbox"
            tabIndex={0}
          >
            <li
              className="ag ae bc ai bd"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <div
                className="be bf bg bh bi bj bk bl bm bn bo bp ag ae bq br bs bt bu"
              >
                <div
                  className="bv"
                  data-testid="employee-record"
                >
                  <span
                    className="bw bx by"
                  >
                    john dow
                  </span>
                  <span
                    className="bz"
                  >
                    Age: 18 years
                    , 
                    Employee Id: Emp123
                  </span>
                </div>
                <div
                  className="ag ae"
                >
                  <svg
                    className="c0 c1 c2 c3 c4"
                    data-baseweb="icon"
                    viewBox="0 0 24 24"
                  >
                    <title>
                      Chevron Right
                    </title>
                    <path
                      clipRule="evenodd"
                      d="M9.29289 7.29289C8.90237 7.68342 8.90237 8.31658 9.29289 8.70711L12.5858 12L9.29289 15.2929C8.90237 15.6834 8.90237 16.3166 9.29289 16.7071C9.68342 17.0976 10.3166 17.0976 10.7071 16.7071L14.7071 12.7071C14.8946 12.5196 15 12.2652 15 12C15 11.7348 14.8946 11.4804 14.7071 11.2929L10.7071 7.29289C10.3166 6.90237 9.68342 6.90237 9.29289 7.29289Z"
                      fillRule="evenodd"
                    />
                  </svg>
                </div>
              </div>
            </li>
          </ul>
        </div>
        <div
          className="c5 ae au"
        >
          <div
            className="bz ai"
            data-testid="employee-addresses"
          >
            <div
              className="bz ai"
            >
              <div
                className="bz af ag c7 ai c8"
              >
                <div
                  className=""
                  data-baseweb="block"
                >
                  <b>
                    john dow's
                  </b>
                   Addresses are following:
                </div>
              </div>
              <ul
                className="ai c6 bz b3 b4 ax"
              >
                <li
                  className="ag av ae bc ai"
                >
                  <div
                    className="be bf bg bh bi bj bk bl bm bn bo bp ag ae bq br bs bt bu"
                  >
                    <div>
                      <div
                        className="am an ao bx c9"
                        data-baseweb="typo-labelmedium"
                      >
                        <span
                          data-testid="employee-address-location"
                        >
                          Kolkata
                        </span>
                      </div>
                    </div>
                  </div>
                </li>
                <li
                  className="ag av ae bc ai"
                >
                  <div
                    className="be bf bg bh bi bj bk bl bm bn bo bp ag ae bq br bs bt bu"
                  >
                    <div>
                      <div
                        className="am an ao bx c9"
                        data-baseweb="typo-labelmedium"
                      >
                        <span
                          data-testid="employee-address-location"
                        >
                          Delhi
                        </span>
                      </div>
                    </div>
                  </div>
                </li>
                <li
                  className="ag av ae bc ai"
                >
                  <div
                    className="be bf bg bh bi bj bk bl bm bn bo bp ag ae bq br bs bt bu"
                  >
                    <div>
                      <div
                        className="am an ao bx c9"
                        data-baseweb="typo-labelmedium"
                      >
                        <span
                          data-testid="employee-address-location"
                        >
                          Bombay
                        </span>
                      </div>
                    </div>
                  </div>
                </li>
                <li
                  className="ag av ae bc ai"
                >
                  <div
                    className="be bf bg bh bi bj bk bl bm bn bo bp ag ae bq br bs bt bu"
                  >
                    <div>
                      <div
                        className="am an ao bx c9"
                        data-baseweb="typo-labelmedium"
                      >
                        <span
                          data-testid="employee-address-location"
                        >
                          Hydrabad
                        </span>
                      </div>
                    </div>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className=""
  />
</section>
`;
